app/db/interface.py

```diff
@@ -71,8 +75,33 @@ def update_payment(self, payment_id, payment_replacement=None, payment_attribute
         :param payment_replacement: <dict> a complete payload that describes a new payload which replaces the old one
         :param payment_attributes: <dict> a collection of new payment attribute values that will overwrite old ones
         """
-
-        raise NotImplementedError()
+        print 'hhhhhhh'
+        payment = self.db.session.query(Payment).get(payment_id)
+        print 'kkkk'
+        if payment == None or payment.is_removed == True :
+            print 'potato'
+            raise InvalidPaymentID('Invalid payment: Payment ID not found',status_code=404)
+        if not payment_replacement and not payment_attributes:
+            raise DataValidationError('Invalid payment: body of request contained bad or no data')
+        if payment_replacement:
+            print 'jjjjjj'
+            # TODO 	: test cases for validity in next sprint
+            if not self.is_valid_put(payment.serialize(),payment_replacement):
+                print 'not valid'
+                raise DataValidationError('Invalid payment: body of request contained bad or no data')
+            print 'lllllllll'
+            payment.deserialize_put(payment_replacement)
+        elif payment_attributes: #patch
+            existing_payment = payment.serialize()
```

Comment:
we would also need to validate `payment_attributes` data in the same way as `payment_replacement`, but more on this below in is_valid function